transmart-core-db-tests/src/main/groovy/org/transmartproject/db/StudyTestData.groovy

```diff
@@ -36,11 +36,15 @@ class StudyTestData {
                 if(!candidates) {
                     assert DimensionImpl.isBuiltinDimension(it), "Unknown dimension name '$it', " +
                             "modifier dimensions as string are not supported in createStudy()"
-                    study.addToDimensionDescriptions(new DimensionDescription(name: it))
+                    // saving is necessary here so that this dimension will be visible to other studies that are
+                    // created within this same transaction. Without it we will have duplicate dimensions.
+                    def dim = new DimensionDescription(name: it).save()
```

Comment:
Hm. I don't really understand this code.
Running below code for two different studies must try to produce 2 different dimension descriptions with the same name and then fail because of the constraint.
`new DimensionDescription(name: 'study').save()`